fairlearn/reductions/_moments/weighted_error_rate.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
+import pandas as pd
+from .moment import ClassificationMoment, _ALL, _LABEL
+_WEIGHTS = "weights"
+
+
+class WeightedErrorRate(ClassificationMoment):
+    r"""Weighted Loss error.
+
+    Parameters
+    ----------
+    loss : {SquareLoss, AbsoluteLoss}
+        A loss object with an `eval` method, e.g. `SquareLoss` or
+        `AbsoluteLoss`.
+    """
+
+    short_name = "Weighted Error Rate"
+
+    def __init__(self):
+        super(WeightedErrorRate, self).__init__()
+
+    '''for what we need here is augmented data. Hence to avoid unnecessary calculation, we use
+    augmented data having been calculated in regression_moment here and directly return in
+    the function load_data to suit the inferface of _lagrangian.'''
```

Comment:
typo: inferface

Also please don't use `'''` for comments. `#` is fine, and we can do multiple lines in a row with `#` as well.

This class doesn't seem related to regression_moment, so I'm a bit confused about what this comment is trying to tell me.

Reply:
I think it is related to the regression_moment, for the init function for CDF_DemographicParity contains code `self.objective = WeightedErrorRate()`. The _lagrangian passes the original x and y to the WeightedErrorRate, but what it really needs is the augmented version. I consider it is a little bit awkward if I recalculate the augmented version in the WeightedErrorRate, which means that this class should store a copy of the grids and loss.